libnavigation-core/src/main/java/com/mapbox/navigation/core/directions/session/RoutesObserver.kt

```diff
@@ -0,0 +1,18 @@
+package com.mapbox.navigation.core.directions.session
+
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+
+/**
+ * Interface definition for an observer that get's notified whenever a list of maintained routes changes.
+ */
```

Comment:
Can we add what might lead to route changes? I am guessing if a user requests a new route himself by changing the destination this won't be invoked. This looks like case when `faster route` for instance changes the routes for you. Similarly, what other factors might lead to this being invoked shall be a good addition here.

Reply:
Whatever is returned in the `RoutesRequestCallback#onRoutesReady` or whenever a user calls `#setRoutes` directly is going to invoke this callback. This gives us internally an option to track those changes and update Nav Native, but also expose a general observable object for the user (or our UI lib) to listen for a change in the primary route and act accordingly (update UI, emit a voice message, etc.).